### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by weilingooi**

### CODE REVIEW

**Comment by weilingooi** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
-  before_action :check_permitted_or_not , only: %i(edit destroy)
+  before_action :check_permitted_or_not, only: %i(edit destroy)
+
+  def new
+    @course = Course.new
+  end
 
   def index
-    @courses = Course.recent_courses.page params[:page]
+    @course = Course.recent_courses.page params[:page]
```
Comment (saobv-2223) [inline app/controllers/courses_controller.rb]: add records per page and dont forget to move 5 to settings file and call it herer
```suggestion
    @course = Course.recent_courses.page(params[:page]).per 5
```

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
 
-  def new
-    @course = Course.new
-  end
-
   def edit; end
 
   def create
-    @course = Course.new　course_params
+    @course = Course.new　(course_params)
```
Comment (saobv-2223) [inline app/controllers/courses_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,11 +1,16 @@
 class UsersController < ApplicationController
   before_action :set_user, expect: %i(:new, :create, :index)
+  before_action :authenticate_user! , only: %i(:edit, :update, :destroy)
+  before_action :check_permitted_or_not , only: %i(:edit, :update)
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  before_action :authenticate_user! , only: %i(edit update destroy)
  before_action :check_permitted_or_not , only: %i(edit update)
```

**Comment by weilingooi** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 4]
```diff
@@ -1,18 +1,17 @@
...
-  before_action :check_permitted_or_not , only: %i(edit destroy)
+  before_action :check_permitted_or_not, only: %i(edit destroy)
+
+  def new
+    @course = Course.new
+  end
 
   def index
-    @courses = Course.recent_courses.page params[:page]
+    @course = Course.recent_courses.page(params[:page]).per 5
```
Comment (saobv-2223) [inline app/controllers/courses_controller.rb]: move 5 to settings file and call it here

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,11 +1,16 @@
...
-  before_action :authenticate_user! , only: %i(:edit, :update , :destroy)
-
-  before_action :check_permitted_or_not , only: %i(:edit , :destroy)
+  def new
+    @user = User.new
+  end
 
-  def index; end
+  def index
+    @users = User.order(:name).page params[:page]
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: add per page

File: app/javascript/packs/application.js [hunk 6]
```diff
@@ -15,3 +17,6 @@ require('channels')
 //
 // const images = require.context('../images', true)
 // const imagePath = (name) => images(name, true)
+
+require("trix")
+require("@rails/actiontext")
```
Comment (saobv-2223) [inline app/javascript/packs/application.js]: using '

**Comment by weilingooi** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 7]
```diff
@@ -45,26 +44,24 @@ def update
...
         format.json { head :no_content }
+      end
     end
   end
 
   private
 
   def course_params
-    params.require :course.permit :title,:user_id
+    params.require (:course).permit :title, :description, :started_at, :member, :status, :total_month
```
Comment (manhnt-0960) [inline app/controllers/courses_controller.rb]:
```suggestion
    params.require(:course).permit :title, :description, :started_at, :member, :status, :total_month
```

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -85,4 +86,8 @@ def authenticate_user!
       render json: { status: 401 }
     end
   end
-end
+
+  def admin_user
+    redirect_to(root_url) unless current_user.admin?
+    end
```
Comment (manhnt-0960) [inline app/controllers/users_controller.rb]: let's check indent

File: app/controllers/courses_controller.rb [hunk 9]
```diff
@@ -45,26 +44,24 @@ def update
...
   end
 
   def set_course
-    @course = Course.find_by(params[:user_id])
+    @course = Course.find_by id: params[:id]
     return if @course
-
-    flash[:warning] = t"not_found"
-    redirect_to root_path
```
Comment (manhnt-0960) [inline app/controllers/courses_controller.rb]: why did you remove it ?

**Comment by weilingooi** [comment]
ready

**Comment by weilingooi** [comment]
ready
